CONTRIBUTING.md

```diff
@@ -1,15 +1,158 @@
+ branch.
 
+### Test against Wildfly
 To run all tests against WildFly (takes about 1 hour):
 
-    mvn clean verify -Dappserver=wildfly8 -DstaticAnalysis -Dchromefirefox
+mvn clean verify -Dappserver=wildfly8 -DstaticAnalysis -Dchromefirefox
+
+### <a name="Commit_Message_Format"> Commit Message Format</a>
+We are inspired the [angular commit message format]
+(https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit),
+that is:
+
+  Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+#### Type
+Must be one of the following:
+
+- **feat**: A new feature
+- **fix**: A bug fix
+- **docs**: Documentation only changes
+- **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing
+             semi-colons, etc)
+- **refactor**: A code change that neither fixes a bug nor adds a feature
+- **perf**: A code change that improves performance
+- **test**: Adding missing tests
+- **chore**: Changes to the build process or auxiliary tools and libraries such as documentation
+generation
+- **revert**: reverts a previous commit. The subject should be the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+#### Scope
+The scope could be anything specifying place of the commit change. For example `$location`,
+`$browser`, `$compile`, `$rootScope`, `ngHref`, `ngClick`, `ngView`, etc...
+
+#### Subject
+The subject contains succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* do not capitalize first letter
+* no dot (.) at the end
+
+#### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+#### Footer
+The footer should contain any information about **Breaking Changes** and is also the place to
+reference GitHub issues that this commit **Closes**.
+
+**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.
+
+### Submit a Pull Request
+
+
+1. Make your changes in a new git branch:
+
+```shell
+git checkout -b my-pr-branch master
+```
+2. Create your patch, **including appropriate test cases**.
+3. Commit your changes using a descriptive commit message that follows our
+    [commit message format](#Commit_Message_Format).
+
+```shell
+git commit -a
```

Comment:
I do not recommend use of `git commit -a` since it can accidentally include changes the user doesn't mean to include. I think we should just assume most contributors will know how to make commits with git using command lin or gui tools. We should just link to a guide about it for the people who don't know how to use git.